Panic when using TabPage

I'm getting this error when trying to debug a walk app in VSCode on Windows 11.
```
Build Error: go build -o c:\Users\User\go\pkg\mod\github.com\tailscale\win@v0.0.0-20240926211701-28f7e73c7afb\__debug_bin3166877821.exe -gcflags all=-N -l .
win.go:14:2: golang.org/x/exp@v0.0.0-20230127140709-cafedaf64729: missing go.sum entry for go.mod file; to add it:
	go mod download golang.org/x/exp (exit status 1)
```
Maybe you need to run `go mod tidy`?

The reason I need to debug it is the call to InitApp() is failing with this error: `InitCommonControlsEx failed`

The build error was because I hadn't configured a launch profile in VSCode.
The InitCommonControlsEx call succeeded after I updated the version number in the manifest, used the rsrc tool as recommended, and ran VSCode as administrator.

Now I'm getting a panic that I think is related to the TabWidget.
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x40 pc=0xba049a]

goroutine 1 [running, locked to thread]:
github.com/tailscale/walk.(*WidgetBase).assignCtrlIDs(0xc000005888)
        C:/Users/User/go/pkg/mod/github.com/tailscale/walk@v0.0.0-20241119233358-1f55a1e15a23/widget.go:393 +0x1a
github.com/tailscale/walk.(*WidgetBase).init(0xc000005888, {0xc80f00?, 0xa4b54b?})
        C:/Users/User/go/pkg/mod/github.com/tailscale/walk@v0.0.0-20241119233358-1f55a1e15a23/widget.go:139 +0x27
github.com/tailscale/walk.(*WindowBase).finishInit(0xc000005888)
        C:/Users/User/go/pkg/mod/github.com/tailscale/walk@v0.0.0-20241119233358-1f55a1e15a23/window.go:740 +0xe5
github.com/tailscale/walk.defaultWndProc(0x11051c, 0x1, 0x0, 0xf2e87fef60)
        C:/Users/User/go/pkg/mod/github.com/tailscale/walk@v0.0.0-20241119233358-1f55a1e15a23/window.go:2294 +0x1e5
syscall.Syscall12(0xc9927e?, 0x1c?, 0xc000010900?, 0xffffffffffffffff?, 0x9f09a5?, 0xc00010a4a8?, 0x9ee4ff?, 0xc71060?, 0x1d?, 0xffffffff80000000, ...)
        C:/Program Files/Go/src/runtime/syscall_windows.go:475 +0x85
github.com/tailscale/win.CreateWindowEx(0x10000, 0x9f4eb3?, 0x212948ac9a8?, 0x54000000, 0x80000000, 0x80000000, 0x80000000, 0x80000000, 0x13045c, 0x0, ...)
        C:/Users/User/go/pkg/mod/github.com/tailscale/win@v0.0.0-20240926211701-28f7e73c7afb/user32.go:2287 +0x125
github.com/tailscale/walk.initWindowWithCfg(0xc00005a3c0)
        C:/Users/User/go/pkg/mod/github.com/tailscale/walk@v0.0.0-20241119233358-1f55a1e15a23/window.go:668 +0x32c
github.com/tailscale/walk.InitWindow(...)
        C:/Users/User/go/pkg/mod/github.com/tailscale/walk@v0.0.0-20241119233358-1f55a1e15a23/window.go:595
github.com/tailscale/walk.InitWidget({0xcf7720, 0xc000005888}, {0xcf1fc0, 0xc00013a408}, {0xc9927e, 0x1c}, 0x50000000, 0x10000)
        C:/Users/User/go/pkg/mod/github.com/tailscale/walk@v0.0.0-20241119233358-1f55a1e15a23/widget.go:117 +0x116
github.com/tailscale/walk.NewCompositeWithStyle({0xcf1fc0, 0xc00013a408}, 0x0)
        C:/Users/User/go/pkg/mod/github.com/tailscale/walk@v0.0.0-20241119233358-1f55a1e15a23/composite.go:30 +0xbc
github.com/tailscale/walk/declarative.Composite.Create({{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...}, ...)
        C:/Users/User/go/pkg/mod/github.com/tailscale/walk@v0.0.0-20241119233358-1f55a1e15a23/declarative/composite.go:69 +0x5d
github.com/tailscale/walk/declarative.(*Builder).InitWidget(0xc000138000, {0xceb988, 0xc00012eb08}, {0xcf1fc0, 0xc00013a408}, 0xc00010b130)
        C:/Users/User/go/pkg/mod/github.com/tailscale/walk@v0.0.0-20241119233358-1f55a1e15a23/declarative/builder.go:383 +0x1cdf
github.com/tailscale/walk/declarative.TabPage.Create({{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...}, ...)
        C:/Users/User/go/pkg/mod/github.com/tailscale/walk@v0.0.0-20241119233358-1f55a1e15a23/declarative/tabpage.go:72 +0xb0
github.com/tailscale/walk/declarative.TabWidget.Create.func1()
        C:/Users/User/go/pkg/mod/github.com/tailscale/walk@v0.0.0-20241119233358-1f55a1e15a23/declarative/tabwidget.go:79 +0x205
github.com/tailscale/walk/declarative.(*Builder).InitWidget(0xc000138000, {0xceb808, 0xc0000738c8}, {0xcf16f0, 0xc00013a008}, 0xc00010bc38)
        C:/Users/User/go/pkg/mod/github.com/tailscale/walk@v0.0.0-20241119233358-1f55a1e15a23/declarative/builder.go:410 +0x1f47
github.com/tailscale/walk/declarative.TabWidget.Create({{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...}, ...)
        C:/Users/User/go/pkg/mod/github.com/tailscale/walk@v0.0.0-20241119233358-1f55a1e15a23/declarative/tabwidget.go:68 +0xbf
github.com/tailscale/walk/declarative.(*Builder).InitWidget(0xc000138000, {0xceb9a8, 0xc000073688}, {0xcf25a0, 0xc000136008}, 0xc00010c8b8)
        C:/Users/User/go/pkg/mod/github.com/tailscale/walk@v0.0.0-20241119233358-1f55a1e15a23/declarative/builder.go:383 +0x1cdf
github.com/tailscale/walk/declarative.MainWindow.Create({{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
        C:/Users/User/go/pkg/mod/github.com/tailscale/walk@v0.0.0-20241119233358-1f55a1e15a23/declarative/mainwindow.go:131 +0x736
main.main()
        C:/Users/User/source/repos/MyProject/main.go:64 +0xadc
```

I've traced it down to this function in `window.go`:
```go
func (wb *WindowBase) ancestor() Form {
	hWndRoot := win.GetAncestor(wb.Handle(), win.GA_ROOT)
	rw, _ := windowFromHandle(hWndRoot).(Form)
	return rw
}
```
The cast to `Form` is not succeeding because the window is a `walk.TabPage` which does not implement Form, but the ok value is not checked. This should be changed to:
```
func (wb *WindowBase) ancestor() Form {
	hWndRoot := win.GetAncestor(wb.Handle(), win.GA_ROOT)
	rw, ok := windowFromHandle(hWndRoot).(Form)
        if !ok {
            return nil
        }
	return rw
}
```
Then additional error handling needs to go in `widget.go`:
```go
func (wb *WidgetBase) assignCtrlIDs() {
	form := wb.ancestor()
        // if form == nil {
        // ...
        // }
	form.AsFormBase().assignCtrlIDs(wb)
}
```

--- a/widget.go
+++ b/widget.go
@@ -389,13 +389,15 @@ func (wb *WidgetBase) SetParent(parent Container) (err error) {
 }
 
 func (wb *WidgetBase) assignCtrlIDs() {
-	form := wb.ancestor()
-	form.AsFormBase().assignCtrlIDs(wb)
+	if form := wb.ancestor(); form != nil {
+		form.AsFormBase().assignCtrlIDs(wb)
+	}
 }
 
 func (wb *WidgetBase) revokeCtrlIDs() {
-	form := wb.ancestor()
-	form.AsFormBase().revokeCtrlIDs(wb)
+	if form := wb.ancestor(); form != nil {
+		form.AsFormBase().revokeCtrlIDs(wb)
+	}
 }
 
 func (wb *WidgetBase) ForEachAncestor(f func(window Window) bool) {

